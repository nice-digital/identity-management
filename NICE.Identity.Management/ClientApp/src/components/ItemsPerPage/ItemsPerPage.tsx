import React from "react";
import { useHistory, useLocation } from "react-router-dom";
import { removeQueryParameter } from "../../utils/querystring";

type ItemsPerPageProps = {
	amount: number | string;
};

export const ItemsPerPage = (props: ItemsPerPageProps): React.ReactElement => {
	const { amount } = props;
	const history = useHistory();
	const { search: querystring, pathname } = useLocation();

	return (
		<div className="mr--e" data-qa-sel="number-of-results-on-page">
			<label htmlFor="itemsPerPage" className="bold mr--c">
				Items per page
			</label>
			<select
				id="itemsPerPage"
				name="itemsPerPage"
				onChange={(event) => {
					const qs = new URLSearchParams(querystring);
					qs.set("amount", event.target.value);
					history.push({
						search: qs.toString(),
					});
				}}
				value={amount}
				data-qa-sel="result-on-the-page-index"
			>
				<option value="25">25</option>
				<option value="50">50</option>
				<option value="all">All</option>
			</select>
		</div>
	);
};
