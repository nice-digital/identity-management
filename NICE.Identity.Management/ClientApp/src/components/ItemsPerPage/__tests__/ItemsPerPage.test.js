import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { ItemsPerPage } from "../ItemsPerPage";

test("should match the snapshot when querystring amount is all", async () => {	
	const {container} = render(
		<MemoryRouter initialEntries={['/organisations?amount=all']}>
			<ItemsPerPage />
		</MemoryRouter>
	);	
	expect(container).toMatchSnapshot();
});