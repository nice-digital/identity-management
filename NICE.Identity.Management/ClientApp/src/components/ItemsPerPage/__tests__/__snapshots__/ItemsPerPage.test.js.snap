// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot when querystring amount is all 1`] = `
<div>
  <div
    class="mr--e"
    data-qa-sel="number-of-results-on-page"
  >
    <label
      class="bold mr--c"
      for="itemsPerPage"
    >
      Items per page
    </label>
    <select
      data-qa-sel="result-on-the-page-index"
      id="itemsPerPage"
      name="itemsPerPage"
    >
      <option
        value="25"
      >
        25
      </option>
      <option
        value="50"
      >
        50
      </option>
      <option
        value="all"
      >
        All
      </option>
    </select>
  </div>
</div>
`;
